src/interface/sparse_pack.hpp

```diff
@@ -255,14 +255,27 @@ class SparsePack : public SparsePackBase {
   }
 
   // operator() overloads
-  KOKKOS_INLINE_FUNCTION
-  auto &operator()(const int b, const int idx) const { return pack_(0, b, idx); }
+  using TE = TopologicalElement;
+  KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const TE el, const int idx) const {
+    return pack_(static_cast<int>(el) % 3, b, idx);
+  }
+  KOKKOS_INLINE_FUNCTION auto &operator()(const int b, const int idx) const {
+    return pack_(0, b, idx);
```

Comment:
Can there be a `PARTHENON_DEBUG_REQUIRE(pack_.GetDim(0) == 1, "Suppressed element index for multi-valued topological element");` here? I can imagine the wrong choice of `&operator()` here being an easy mistake that's hard to debug. (I'm not 100% sure that `GetDim(0)` is the right index)

Reply:
You are right, we need some kind of check here. I think it probably should be something like 
```c++
PARTHENON_DEBUG_REQUIRE(pack_(0, b, idx).topological_type == TopologicalType::Cell, "Suppressed topological element index assumes that this is a cell variable, but it isn't");
```